casper/src/main/scala/io/casperlabs/casper/util/CasperLabsProtocol.scala

```diff
@@ -0,0 +1,97 @@
+package io.casperlabs.casper.util
+
+import cats.Applicative
+import cats.implicits._
+import io.casperlabs.casper.consensus.{state, Block}
+import io.casperlabs.casper.util.ProtocolVersions.Config
+import io.casperlabs.catscontrib.MonadThrowable
+import io.casperlabs.ipc
+import io.casperlabs.ipc.ChainSpec.DeployConfig
+import simulacrum.typeclass
+
+@typeclass
+trait CasperLabsProtocol[F[_]] {
+
+  /** Returns a [[state.ProtocolVersion]] at block height. */
+  def versionAt(blockHeight: Long): F[state.ProtocolVersion]
+
+  /** Returns a [[state.ProtocolVersion]] at specific block. */
+  def protocolFromBlock(b: Block): F[state.ProtocolVersion]
+
+  /** Returns a configuration at specific block height.
+    *
+    * Note that this "merges" all configurations up to that block height.
+    * Specifically if certain conifguration parameter isn't defined at latest
+    * upgrade point, latest one will be used.
+    */
+  def configAt(blockHeight: Long): F[Config]
+}
+
+object CasperLabsProtocol {
+  import ProtocolVersions.Config
+
+  def unsafe[F[_]: Applicative](
+      versions: (Long, state.ProtocolVersion, Option[ipc.ChainSpec.DeployConfig])*
+  ): CasperLabsProtocol[F] = {
+    def toDeployConfig(ipcDeployConfig: Option[ipc.ChainSpec.DeployConfig]): DeployConfig =
+      ipcDeployConfig
+        .map(d => DeployConfig(d.maxTtlMillis, d.maxDependencies))
+        .getOrElse(DeployConfig())
+
+    def toConfig(
+        blockHeightMin: Long,
+        protocolVersion: state.ProtocolVersion,
+        deployConfig: Option[ipc.ChainSpec.DeployConfig]
+    ): Config =
+      Config(blockHeightMin, protocolVersion, toDeployConfig(deployConfig))
+
+    val underlying = ProtocolVersions(versions.map {
+      case (rank, protocolVersion, ipcDeployConfig) =>
+        Config(rank, protocolVersion, toDeployConfig(ipcDeployConfig))
```

Comment:
```suggestion
        toConfig(rank, protocolVersion, toDeployConfig(ipcDeployConfig))
```
Or is `toConfig` needed at all?